books/models.py

```diff
@@ -147,3 +149,16 @@ def __unicode__(self):
     @models.permalink
     def get_absolute_url(self):
         return 'pathagar.books.views.book_detail', [self.pk]
+
+
+@receiver(post_delete, sender=Book)
+def book_post_delete_handler(**kwargs):
+    book = kwargs['instance']
+
+    book_storage, book_file = book.book_file.storage, book.book_file.path
```

Comment:
Actually, I believe you don't really have to get the storage for doing this: it should be enough to make something like:
```python
if book.book_file:
    book.book_file.delete()
```
As basically it will automatically call the storage internally, and make the code simpler! (Same for the `cover_img` field).